### PR TITLE
Master datagrid row overlay

### PR DESCRIPTION
**Pull request by stather**
Adds 2 new event callback to datagrid, and a feature to display an overlay when the mouse is over the row.

New Parameter RowOver of type EventCallback<DataGridRowMouseEventArgs<TItem>>
This is called when the mouse is over the row.

New Parameter RowLeave of type EventCallback<DataGridRowMouseEventArgs<TItem>>
This is called when the mouse leaves a row.

These are useful anyway, but used mainly to facilitate the row overlay feature.

New Parameter OverlayTemplate of type RenderFragment<TItem>.

This render fragment can be used to display additional information about the row item when the mouse is over the row. The content of the fragment is displayed over the top of the current row and is right aligned.

```
<DataGrid TItem="sometype">
                       <OverlayTemplate>
                            <Div Flex="Flex.JustifyContent.End.AlignItems.Center" Margin="Margin.Is3.FromBottom" Style="width:400px;background-color:white;">
                                <Div Padding="Padding.Is2">
                                    <Button Size="Size.Small" Color="Color.Secondary" onclick="event.stopPropagation();">Ack</Button>
                                </Div>
                                <Div Padding="Padding.Is2">
                                    <Button Size="Size.Small" Color="Color.Secondary" onclick="event.stopPropagation();">Close</Button>
                                </Div>
                                <Div Padding="Padding.Is2" @onclick:stopPropagation="true" Overflow="Overflow.Visible">
                                    <Dropdown Style="z-index:2000;">
                                        <DropdownToggle Size="Size.Small" Clicked="ToggleClicked">
                                            ...
                                        </DropdownToggle>
                                        <DropdownMenu>
                                            <DropdownItem>Add responder</DropdownItem>
                                            <DropdownItem>Delete</DropdownItem>
                                        </DropdownMenu>
                                    </Dropdown>
                                </Div>
                            </Div>
                        </OverlayTemplate>
</DataGrid>
```

The DataGrid only displays this fragment when an OverlayTemplate is present and the mouse is over a row. 
The fragment is removed when the mouse leaves the row.

### CODE REVIEW

**Comment by stather** [comment]
Added emails to account

**Comment by stather** [comment]
recheck

**Comment by stather** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by stather** [comment]
recheck

File: Source/Extensions/Blazorise.DataGrid/Blazorise.DataGrid.csproj [hunk 1]
```diff
@@ -9,6 +9,7 @@
   <PropertyGroup>
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
     <NoWarn>1591</NoWarn>
+    <Version>1.1.3</Version>
```
Comment (stsrki) [inline Source/Extensions/Blazorise.DataGrid/Blazorise.DataGrid.csproj:12]: We control versions from a `Blazorise.props` file. You can safely remove this line.

File: Source/Blazorise/Components/Table/TableRow.razor.cs [hunk 2]
```diff
@@ -111,6 +131,16 @@ public Cursor HoverCursor
         /// </summary>
         [Parameter] public EventCallback<BLMouseEventArgs> DoubleClicked { get; set; }
 
+        /// <summary>
+        /// Occurs when the row is mouse overed.
+        /// </summary>
+        [Parameter] public EventCallback<BLMouseEventArgs> MouseOvered { get; set; }
```
Comment (stsrki) [inline Source/Blazorise/Components/Table/TableRow.razor.cs:137]: Rename to `MouseOver`.

File: Source/Extensions/Blazorise.DataGrid/_DataGridRow.razor [hunk 3]
```diff
@@ -64,6 +66,15 @@
                     }
                 </TableRowCell>
             }
+        }        
+        @if (ParentDataGrid.OverlayTemplate != null && mouseIsOver)
+        {
+            <TableRowCell Style="width:0px;margin:0px;padding:0px" Overflow="Overflow.Visible">
+                <Div Style="position:absolute;right:0px;margin:0px;padding;0px">
```
Comment (stsrki) [inline Source/Extensions/Blazorise.DataGrid/_DataGridRow.razor:73]: Prefer Blazorise utilities over inline styles

- Position
- Margin
- Padding

File: .cr/personal/FavoritesList/List.xml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (stsrki) [inline .cr/personal/FavoritesList/List.xml:1]: I'm not sure what is `.cr/personal/FavoritesList/List.xml` but consider removing it or add it to a `.gitignore`

File: Source/Blazorise/Components/Table/TableRow.razor [hunk 5]
```diff
@@ -2,6 +2,8 @@
 @inherits BaseDraggableComponent
 <tr id="@ElementId" class="@ClassNames" style="@StyleNames" @onclick="@OnClickHandler"
     draggable="@DraggableString"
+    @onmouseover="@OnMouseOverHandler"
```
Comment (stsrki) [inline Source/Blazorise/Components/Table/TableRow.razor:5]: Do we need `:preventDefault` for these events?

File: Source/Extensions/Blazorise.DataGrid/Blazorise.DataGrid.csproj (hunk 1 above)
Comment (stather) [reply Source/Extensions/Blazorise.DataGrid/Blazorise.DataGrid.csproj]: Removed line

File: Source/Blazorise/Components/Table/TableRow.razor.cs (hunk 2 above)
Comment (stather) [reply Source/Blazorise/Components/Table/TableRow.razor.cs]: Changed to MouseOver

File: Source/Extensions/Blazorise.DataGrid/_DataGridRow.razor (hunk 3 above)
Comment (stather) [reply Source/Extensions/Blazorise.DataGrid/_DataGridRow.razor]: Changed to use utilities

File: .cr/personal/FavoritesList/List.xml (hunk 4 above)
Comment (stather) [reply .cr/personal/FavoritesList/List.xml]: Deleted file and added to .gitignore

File: Source/Blazorise/Components/Table/TableRow.razor (hunk 5 above)
Comment (stather) [reply Source/Blazorise/Components/Table/TableRow.razor:5]: It doest seem like it

**Comment by stather** [comment]
recheck

**Comment by stather** [comment]
@stsrki Hi, I've made the changes. I cannot see for the life of me why the CLA check is failing. Any ideas?

**Comment by stsrki** [comment]
`OverlayTemplate` might be a too generic name. My first guess was that it covers the entire of the DataGrid area. Better naming would be a `RowOverlayTemplate`. Any other ideas? 

Also might be a good idea to be able to define overlay position, Start or End of the row.

**Comment by stather** [comment]
Renamed parameter to RowOverlayTemplate.
Added Parameter RowOverlayPosition to specify overlay should be positioned at start or end of row.
Added corresponding enum for RowOverlayPosition

File: Source/Extensions/Blazorise.DataGrid/Enums/RowOverlayPosition.cs [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    /// <summary>
+    /// Defines the position of the row overlay
+    /// </summary>
+    public enum RowOverlayPosition
+    {        
+        /// <summary>
+        /// Overlay start at left of row.
+        /// </summary>
+
+        Left,
```
Comment (stsrki) [inline Source/Extensions/Blazorise.DataGrid/Enums/RowOverlayPosition.cs:12]: Prefer naming `Start` and `End` instead of `Left` and `Right`.

Comment (stather) [reply Source/Extensions/Blazorise.DataGrid/Enums/RowOverlayPosition.cs]: Renamed as requested

Comment (stsrki) [reply Source/Extensions/Blazorise.DataGrid/Enums/RowOverlayPosition.cs]: You're quick :)

**Comment by stsrki** [comment]
Hm, it might be. @stather can you check this

image

**Comment by stather** [comment]
Output from git how seems to have the correct details.
```
C:\Users\rstath698\source\repos\operait\Blazorise>git show
commit 5afdd61d32c3e57a5407c01276ed7aa5ccb5cfa5 (HEAD -> master-datagrid-row-overlay, origin/master-datagrid-row-overlay)
Author: stather <russell.stather@mac.com>
Date:   Wed Nov 23 11:16:34 2022 +0200

    Updated terminology to Start and End
```

**Comment by David-Moreira** [comment]
Can you check this one ? Any reason why you closed the PR? Are you redoing it?
image

**Pull request closed without merging**